pajbot/modules/vip_refresh.py

```diff
@@ -0,0 +1,151 @@
+        if msg_id == "no_vips":
+            return []
+
+        if msg_id == "vips_success":
+            if message.startswith("The VIPs of this channel are: "):
+                message = message[30:]  # 36 = length of the above "prefix"
+                return message.split(", ")
+            log.warning(
+                f"Received vips_success NOTICE message, but actual message did not begin with expected prefix. Message was: {message}"
+            )
+
+        return None
+
+    def _on_pubnotice(self, channel, msg_id, message):
+        if channel != self.bot.streamer:
+            return
+
+        vip_logins = self._parse_pubnotice_for_vips(msg_id, message)
+
+    @time_method
+    def _process_vip_logins(self, vip_logins):
+        # Called on the action queue thread, to resolve the logins to user IDs
+        vip_basics = self.bot.twitch_helix_api.bulk_get_user_basics_by_login(vip_logins)
+
+        # filter out invalid/deleted/etc. users
+        vip_basics = [e for e in vip_basics if e is not None]
+
+        with DBManager.create_session_scope() as db_session:
+            db_session.execute(
+                text(
+                    """
+CREATE TEMPORARY TABLE vips(
+    id TEXT PRIMARY KEY NOT NULL,
+    login TEXT NOT NULL,
+    name TEXT NOT NULL
+)
+ON COMMIT DROP"""
+                )
+            )
+
+            db_session.execute(
+                text("INSERT INTO vips(id, login, name) VALUES (:id, :login, :name)"),
+                [basics.jsonify() for basics in vip_basics],
+            )
+
+            # hint to understand this query: "excluded" is a PostgreSQL keyword that referers
+            # to the data we tried to insert but failed (so excluded.login would be equal to :login
+            # if we only had one value for :login)
+            db_session.execute(
+                text(
+                    """
+WITH updated_users AS (
+    INSERT INTO "user"(id, login, name, vips)
+        SELECT id, login, name, TRUE FROM vips
+    ON CONFLICT (id) DO UPDATE SET
+        login = excluded.login,
+        name = excluded.name,
+        vip = TRUE
+    RETURNING id
+)
+UPDATE "user"
+SET
+    vip = FALSE
+WHERE
+    id NOT IN (SELECT * FROM updated_users) AND
+    vip IS TRUE"""
+                )
+            )
+
+        log.info(f"Successfully updated {len(vip_basics)} VIPs")
+
+    def load_commands(self, **options):
+        self.commands["reload"] = Command.multiaction_command(
+            command="reload",
+            commands={
+                "vips": Command.raw_command(
+                    self.update_vip_cmd,
+                    delay_all=120,
+                    delay_user=120,
+                    level=1000,
+                    examples=[
+                        CommandExample(
+                            None,
+                            f"Reload who is a Twitch channel VIP",
+                            chat=f"user:!reload vips\nbot>user: Reloading VIP status...",
```

Comment:
the command as it is right now actually says `Reloading list of VIPs...`, not `Reloading VIP status...`, one of the two should be corrected to be consistent.
I also realize that this same mistake is also in the moderators refresh module, but that can be fixed separately.

Reply:
👍